src/mscorlib/shared/System/IO/BinaryWriter.cs

```diff
@@ -389,6 +390,62 @@ public unsafe virtual void Write(String value)
             }
         }
 
+        public virtual void Write(ReadOnlySpan<byte> value)
+        {
+            if (GetType() == typeof(BinaryWriter))
+            {
+                OutStream.Write(value);
+            }
+            else
+            {
+                int length = value.Length;
+                byte[] buffer = ArrayPool<byte>.Shared.Rent(length);
+                try
+                {
+                    value.CopyTo(buffer);
+                    Write(buffer, 0, length);
+                }
+                finally
+                {
+                    ArrayPool<byte>.Shared.Return(buffer);
+                }
+            }
+        }
+
+        public virtual void Write(ReadOnlySpan<char> value)
+        {
+            if (GetType() == typeof(BinaryWriter))
+            {
+                int length = value.Length;
+                char[] chars = ArrayPool<char>.Shared.Rent(length);
+                try
+                {
+                    value.CopyTo(chars);
+                    byte[] bytes = _encoding.GetBytes(chars, 0, length);
+                    OutStream.Write(bytes);
+                }
+                finally
+                {
+                    ArrayPool<char>.Shared.Return(chars);
+                }
+            }
```

Comment:
At this point, it probably makes sense to just get rid of the if block and only have the else block.  The only difference is whether it calls `OutStream.Write(bytes)` or `Write(bytes)`, and since the latter on the base type just calls the former, we might as well save the extra code (unless we found the extra virtual call mattered performance-wise, but I doubt we would given all the other work that's happening here).